lib/modules/sentiment/entry_point.rb

```diff
@@ -18,6 +18,100 @@ def inject_into(plugin)
 
         plugin.on(:post_created, &sentiment_analysis_cb)
         plugin.on(:post_edited, &sentiment_analysis_cb)
+
+        plugin.add_report("overall_sentiment") do |report|
+          report.modes = [:stacked_chart]
+
+          grouped_sentiments =
+            DB.query(<<~SQL, report_start: report.start_date, report_end: report.end_date)
+            SELECT 
+              DATE_TRUNC('day', p.created_at)::DATE AS posted_at,
+              AVG((cr.classification::jsonb->'positive')::integer) AS avg_positive,
+              -AVG((cr.classification::jsonb->'negative')::integer) AS avg_negative
+            FROM 
+              classification_results AS cr
+            INNER JOIN posts p ON p.id = cr.target_id AND cr.target_type = 'Post'
+            INNER JOIN topics t ON t.id = p.topic_id
+            INNER JOIN categories c ON c.id = t.category_id
+            WHERE 
+              NOT c.read_restricted AND
+              t.archetype = 'regular' AND
+              p.user_id > 0 AND
+              cr.classification_type = 'sentiment' AND
+              (p.created_at > :report_start AND p.created_at < :report_end)
+            GROUP BY DATE_TRUNC('day', p.created_at)
+          SQL
+
+          data_points = %w[positive negative]
+
+          report.data =
+            data_points.map do |point|
+              {
+                req: "sentiment_#{point}",
+                color: point == "positive" ? report.colors[1] : report.colors[3],
+                label: I18n.t("discourse_ai.sentiment.reports.overall_sentiment.#{point}"),
+                data:
+                  grouped_sentiments.map do |gs|
+                    { x: gs.posted_at, y: gs.public_send("avg_#{point}") }
+                  end,
+              }
+            end
+        end
+
+        plugin.add_report("post_emotion") do |report|
+          report.modes = [:radar]
+
+          grouped_emotions =
+            DB.query(<<~SQL, report_start: report.start_date, report_end: report.end_date)
+            SELECT 
+              u.trust_level AS trust_level,
+              AVG((cr.classification::jsonb->'sadness')::integer) AS avg_sadness,
+              AVG((cr.classification::jsonb->'surprise')::integer) AS avg_surprise,
+              AVG((cr.classification::jsonb->'neutral')::integer) AS avg_neutral,
+              AVG((cr.classification::jsonb->'fear')::integer) AS avg_fear,
+              AVG((cr.classification::jsonb->'anger')::integer) AS avg_anger,
+              AVG((cr.classification::jsonb->'joy')::integer) AS avg_joy,
+              AVG((cr.classification::jsonb->'disgust')::integer) AS avg_disgust
+            FROM
+              classification_results AS cr
+            INNER JOIN posts p ON p.id = cr.target_id AND cr.target_type = 'Post'
+            INNER JOIN users u ON p.user_id = u.id
+            INNER JOIN topics t ON t.id = p.topic_id
+            INNER JOIN categories c ON c.id = t.category_id
+            WHERE 
+              NOT c.read_restricted AND
```

Comment:
Will this make the graph empty for sites where all categories are read restricted?  Since peeps seeing this graph will all be staff anyway, I wonder if we should consider all posts 🤔

Reply:
Makes sense 👍